### PR TITLE
Port permission_prompt_tests to Rust

### PR DESCRIPTION
**Pull request by jsouto18**
This PR should close the task `Port permission_prompt_tests to Rust` currently open on #2988, and work towards closing #3129. This tests no longer interact with a ptty (#2767).

### CODE REVIEW

File: cli/tests/integration_tests.rs [hunk 1]
```diff
@@ -1318,6 +1343,32 @@ fn cafile_bundle_remote_exports() {
   drop(g)
 }
 
+#[test]
```
Comment (jsouto18) [inline cli/tests/integration_tests.rs]: @ry Should I modify this test to be able to run them in parallel? Adding the permission, as a suffix, to the function name.

**Comment by ry** [comment]
Note to self, `tools/permission_prompt_test.py` removed in 8f571ef166d31f44378c1b6de5f4092fcccef732 because dead code.

https://github.com/denoland/deno/blob/3c2399e437521122b66a2de735aebbb80a546148/tools/permission_prompt_test.py

File: cli/tests/integration_tests.rs (hunk 1 above)
Comment (ry) [reply cli/tests/integration_tests.rs]: prefix is preferable

Comment (ry) [reply cli/tests/integration_tests.rs]: actually the way you have it is fine. i think it's easier to read this way.

File: cli/tests/permission_test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  netRequired(): void {
+    listen({ transport: "tcp", port: 4541 });
+  },
+  runRequired(): void {
+    run({
+      args: [
+        "python",
+        "-c",
+        "import sys; sys.stdout.write('hello'); sys.stdout.flush()"
```
Comment (hayd) [inline cli/tests/permission_test.ts]: would `Deno.execPath(), eval, "console.log('hello')"` work here?

**Pull request merged**